### ISSUE TITLE
Failed to spawn script 'vpnc-script-win.js' for connect: The specified resource type cannot be found in the image file.

### ISSUE THREAD

**Issue by nielspeen** [openconnect/openconnect-gui#25]
I've been unable to reproduce this yet, but the first time I ran 0.8 I noticed this error in the log:

Failed to spawn script 'vpnc-script-win.js' for connect: The specified resource type cannot be found in the image file.

**Comment by vladr** [comment]
FWIW: simply deleting and re-adding the i/f did not solve the problem for me.  I had to add a SECOND i/f (e.g. Local Area Connection 5) on top of the first one, and then delete the first one (e.g. Local Area Connection 4) using Device Manager for netsh to be able to operate on it and for vpnc-script-win.js to succeed.  One of my suspicions is that at some point in the distant past this machine may have briefly had another "Local Area Connection 4" and that traces of it are still left in the system, confusing netsh's i/f name lookup.

**Comment by tidatida** [comment]
I met the same problem, I 'll recheck it later.

**Comment by nmav** [comment]
Is that still the case with the latest version?

**Comment by TheAndyMac** [comment]
I have this almost every time I try to connect, even with the command line (which comes with the GUI) where I am specifying the interface name specifically. Any further news on a fix for this?

**Comment by nmav** [comment]
@TheAndyMac could there be something "strange" with the path openconnect is installed at? Could you retry with something simpler such as "C:\VPN" ?

**Comment by TheAndyMac** [comment]
Have found the cause (from another forum) which is related to (1) having IPv6 setup on the TAP driver and (2) running Windows 10. Not sure if same problem on pre-Win10 but switching off IPv6 gets rid of the problem (and not an issue for my work VPN as only IPv4 on that anyway). Maybe something in the script itself that does not like IPv6 connectivity on the TAP driver?

**Comment by nmav** [comment]
Could you figure which command fails in vpnc-script.js? E.g. by disabling all IP6 variable setting, and starting enabling few at a time until it fails.

**Comment by TheAndyMac** [comment]
I’ll try to set some time aside to do this in the next few days or so.

**Comment by TheAndyMac** [comment]
I did not manage to do this, but think I have identified one of the issues (at least for the command line level operation using your build). Did have VMWare Player installed to provide the NAT for Hyper-V on my machine. Now that latest Windows 10 build has NAT (albeit without DHCP and only currently via PowerShell) have now removed VMWare Player and configured the Hyper-V NAT network connection – and so far (a number of connections later) not had any issues of the vpnc-script.js failing – so maybe this was something related to the VMWare virtual network ports… Don’t know if this helps any.

**Comment by vkopichenko** [comment]
I had the same error message with latest version under russian windows until manually renamed in ncpa.cpl TAP-Windows Adaptor's network connection name from "Подключение по локальной сети 2" to just "OpenConnect". I guess, the problem was in encoding.

**Comment by spetragl** [comment]
I always have this problem when I re-install. It creates "Local Area Connection 2" which causes openconnect to fail. However if I rename the TAP network to "MyTAP" it works fine. Windows doesn't like embedded blanks in the name.

**Comment by horar** [comment]
@vkopichenko , @spetragl Hi, there was merged a pull request #87 this could solve encoding and spaces in names.

**Comment by spetragl** [comment]
I received this message again today; 
"2016-03-08 10:59 Failed to spawn script 'vpnc-script.js' for connect: There are no more files." which I realize is a little different from above. But I was wondering, can you display the errno / return code from the "exec" after that message? It might give us a clue what's happening. 
I have a suspicion that you're probably receiving a non-zero from the Child (cscript.exe), but the file does indeed exist.

**Comment by mfayez** [comment]
Guys, that did not work for me, that thing that worked for me is to change the file permission of that file:
C:\Program Files\OpenConnect\vpnc-script.js to allow everyone to have full control.

**Comment by horar** [comment]
Hello, guys, could you please re-test with 1.4.0 if this problem is still valid? That script was updated; interfaces are not handled by name and logging is little bit improved (just close-open if you don't see vpnc.log in log window). Just to be update the status... Thanks in advance.

**Comment by TheAndyMac** [comment]
Tested and now seems to be solid. So far, connected first time each time, and connects fully to the AnyConnect servers - when running the command line on the previous version, sometimes it would seem to connect but either not resolve web sites correctly or even connect at all for Outlook/Exchange. With 1.4.0 (at least so far) every connection has been rock solid and complete first time. Awesome work!

**Comment by horar** [comment]
Okay, then closing now - if someone get this kind of problem - feel free to re-open and update info from app's log

**Comment by spetragl** [comment]
I tested your script for v1.4 and the “connect” was flawless.
Even the logfile “vpnc.log” contained all of the activity.
Beautiful!

One last suggestion.
We need to add the code to “disconnect” the custom SPLIT routes cleanly (that were added in “connect”).

Something like this:
  var gw = getDefaultGateway();

  // Delete direct route for the VPN gateway
  echo(">Deleting Direct Route for VPN Gateway");
  exec("route delete " + env("VPNGATEWAY") + " mask 255.255.255.255");

  // Restore direct route
  echo(">Restoring Direct Route");
  exec("route delete 0.0.0.0 mask 0.0.0.0 ");
  exec("route add 0.0.0.0 mask 0.0.0.0 " + gw);

  // ReSet Tunnel Adapter IP = nothing
  echo(">Resetting Tunnel Adapter IP");
  exec("netsh interface ip set address name=\"" + env("TUNDEV") + "\" source=static 1.0.0.0 255.255.255.255");
  exec("netsh interface ip delete address \"" + env("TUNDEV") + "\" 1.0.0.0");

  // Take Down IPv4 Split Tunnel Server-side Network Routes
  if (env("CISCO_SPLIT_INC")) {
      echo(">Removing IPv4 Split Tunnel INC Server-side Network Routes:");
      for (var i = 0 ; i < parseInt(env("CISCO_SPLIT_INC")); i++) {
          var network = env("CISCO_SPLIT_INC_" + i + "_ADDR");
          var netmask = env("CISCO_SPLIT_INC_" + i + "_MASK");
          exec("route delete " + network);
      }
  }

  // Take Down IPv4 Split Tunnel Client-side Network Routes
  if (env("CISCO_SPLIT_LCL")) {
      echo("Removing IPv4 Split Tunnel Local Client-side Network Routes:");
      for (var i = 0 ; i < parseInt(env("CISCO_SPLIT_LCL")); i++) {
          var network = env("CISCO_SPLIT_LCL_" + i + "_ADDR");
          var netmask = env("CISCO_SPLIT_LCL_" + i + "_MASK");
          exec("route delete " + network);
      }
  }
